wagtail_events/abstract_models.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding:utf8 -*-
+
+from __future__ import unicode_literals
+
+from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
+from django.db import models
+from wagtail.wagtailadmin.edit_handlers import FieldPanel
+from wagtail.wagtailcore.models import Page
+
+from wagtail_events.managers import EventInstanceManager
+from wagtail_events.utils import _DATE_FORMAT_RE
+
+
+class AbstractPaginatedIndexPage(Page):
+    """ """
+    paginate_by = models.PositiveIntegerField(blank=True, null=True)
+    content_panels = Page.content_panels + [FieldPanel('paginate_by')]
+
+    class Meta(object):
+        """Django model meta options."""
+        abstract = True
+
+    def _get_children(self, request):
```

Comment:
I'd prefer to keep it as is, this is being overriden in `wagtail_events.models.EventIndexPage`.
Keeping the `_get_children` method on the abstract models makes for less boilerplate to copy when extending. But I defer to the group.